lib/Transport.js

```diff
@@ -1,21 +1,44 @@
 "use strict";
-var Transport = module.exports = function(options) {
-  options = options || { };
-  this._construct(options);
+var Transport = module.exports = function(path, auth) {
+  path = path || { };
+  auth = auth || { };
+  this._construct(path, auth);
 };
 
 var request = require("superagent");
 
-Transport.prototype._construct = function(path) {
+Transport.prototype._construct = function(path, auth) {
   this._path = path;
+  this._auth = {
+    cookie: auth.cookie,
+    header: auth.header
+  };
+};
+
+Transport.prototype._attachAuthToRequest = function(someRequest) {
+  if (this._auth.cookie) {
+    var cookieName = Object.keys(this._auth.cookie)[0];
```

Comment:
It would be possible to allow mutiple cookies and headers to be set. I wouldn't commend this as a wonderful idea, but some auth systems I've seen do this, particularly if they need to encode >4k in a cookie.